### PR TITLE
operator: separate Flink and Spark storage settings

### PR DESCRIPTION
**Pull request by ennru**
### What changes were proposed in this pull request?

Make the default storage claim settings for Flink and Spark independent of each other.

### Why are the changes needed?

The storage claims need to become easier adjustable per application.

### CODE REVIEW

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/DeploymentContext.scala [hunk 1]
```diff
@@ -29,15 +29,13 @@ case class DeploymentContext(kafkaContext: KafkaContext,
...
-                             persistentStorageSettings: PersistentStorageSettings,
                              podName: String,
                              podNamespace: String) {
-  import kafkaContext._
   def infoMessage = s"""
    | pod-name:                         ${podName}
    | pod-namespace                     ${podNamespace}
    |
-   | kafka-bootstrap-servers:          ${bootstrapServers}
+   | kafka-bootstrap-servers:          ${kafkaContext.bootstrapServers}
```
Comment (RayRoestenburg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/DeploymentContext.scala:38]: yeah more readable

File: core/cloudflow-operator/src/main/resources/application.conf [hunk 2]
```diff
@@ -109,6 +109,8 @@ cloudflow {
         # Do not specify max heap, this is calculated by memory-overhead
         java-opts = ""
         java-opts = ${?SPARK_EXECUTOR_JAVA_OPTS}
+
+        persistent-storage = ${cloudflow.platform.deployment.persistent-storage}
```
Comment (RayRoestenburg) [inline core/cloudflow-operator/src/main/resources/application.conf:113]: Maybe this is a good time to make this more consistent with flink-runner config, and have:
```hocon
spark-runner {
  driver {
    # contents of spark-runner-driver
  }
  executor {
    # contents of spark-runner-executor
  }
  persistent-storage = ${cloudflow.platform.deployment.persistent-storage}
}
```

**Review by RayRoestenburg — COMMENTED** [review]
I think it would be nice to fix the conf file slightly to make spark runner settings more consistent with flink settings

**Comment by ennru** [comment]
> I think it would be nice to fix the conf file slightly to make spark runner settings more consistent with flink settings

Agreed, but how would that be backwards compatible?

**Comment by blublinsky** [comment]
Considering that we now allow overwriting of Flink checkpointing through configuration, do we still need this?

**Comment by RayRoestenburg** [comment]
@blublinsky it's not only about Flink of course. https://github.com/lightbend/cloudflow/issues/710 improves on the approach here, the operator should not have storage settings at all.

**Pull request closed without merging**